### DISCUSSION TITLE
Is it possible to send a message from an invoked service using the parent service?

### DISCUSSION THREAD

**Discussion by panesofglass** [statelyai/xstate#1444] [Q&A] [answered]
I'd like to send an event through a parent service to an invoked, child service. Attempting to call `service.send` with a `{to: 'child'}` doensn't seem to work. How can I send an event from an interpreted, parent service to an invoked, child service?

Here's a [repro](https://codesandbox.io/s/crazy-noether-kolcl?file=/src/index.ts) to demonstrate the issue. In the last line, I attempt to kick off the ping-pong cycle by calling the child machine, which is what I'm trying to do above. This doesn't seem to work, and I'm wondering how I might achieve this result.

**Comment by Andarist** [comment, marked as answer]
Please always try to share a repro case in a runnable form. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given (and I wouldn't say that the description is sufficient enough, it has some gaps that I would have to fill in myself).

It's always better to talk having a repro because it ensures that we both can be on the same page - eliminates possible miscommunication problems.

**Reply by davidkpiano** [reply]
For convenience: [CodeSandbox (XState template)](https://codesandbox.io/s/xstate-example-template-qcwed)

**Reply by panesofglass** [reply]
@Andarist and @davidkpiano I thought I had resolved this but discovered that I had just worked around the problem entirely. However, I would very much like to send an event through a parent service to a child. I've created a repro and updated the original question. Thank you for any insight you can offer.

**Reply by Andarist** [reply]
@panesofglass Conceptually you only create the top-most service - the rest is encapsulated and hidden from you. In general, this is the preferred way to handle stuff. You shouldn't talk to your children directly but rather let your root service route an event.

If you really need this then u can grab a named child from your service and talk to it directly but this sort of breaks the encapsulation:
https://codesandbox.io/s/romantic-ramanujan-w6y2f
